src/execution/index/art/node.cpp

```diff
@@ -186,4 +188,208 @@ void Node::Erase(Node *&node, idx_t pos, ART &art) {
 	}
 }
 
+void Node::ResolvePrefixesAndMerge(ART *l_art, ART *r_art, Node *&l_node, Node *&r_node, idx_t depth,
+                                   Node *&l_node_parent, idx_t l_node_pos, Node *&r_node_parent, idx_t r_node_pos) {
+
+	if (!l_node) {
+		l_node = r_node;
+		if (l_node_parent) {
+			l_node_parent->ReplaceChildPointer(l_node_pos, l_node);
+		}
+		if (r_node_parent) {
+			r_node_parent->ReplaceChildPointer(r_node_pos, nullptr);
+		}
```

Comment:
I think this can only happen at the root level, then you never have the parent, maybe add an assertion.